lib/coqProject_file.ml

```diff
@@ -157,6 +134,53 @@ let parse_native ~warning_fn ~error proj flag =
   in
   { proj with native_compiler = Some native }
 
+let escape_char c =
+  match c with
+  | '\'' -> "\'"
+  | '\n' -> "\\n"
+  | '\t' -> "\\t"
+  | c -> String.make 1 c
+
+let check_filename f =
+  let a = ref None in
+  let check_char c =
+    match c with
+    | '\n' | '\t' | '\\' | '\'' | '"' | ' ' | '#' | '$' | '%' -> a := Some c
```

Comment:
This check breaks coq_makefile on Windows, which uses backslash as a file separator.  Windows pathnames may contain `$`, as in `\\wsl$\ ...`.

Much less important: both Linux and Windows let you have filenames with embedded spaces, although almost all developers avoid those.